recipes/qpdf/all/conanfile.py

```diff
@@ -0,0 +1,161 @@
+        tc.variables["USE_IMPLICIT_CRYPTO"] = False
+        tc.cache_variables["CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP"] = True
+        if self.options.with_crypto == "native":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = True
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = False
+        elif self.options.with_crypto == "openssl":
+            tc.variables["REQUIRE_CRYPTO_NATIVE"] = False
+            tc.variables["REQUIRE_CRYPTO_GNUTLS"] = False
+            tc.variables["REQUIRE_CRYPTO_OPENSSL"] = True
+        tc.generate()
+        # TODO: after https://github.com/conan-io/conan/issues/11962 is solved
+        # we might obsolete here cmake deps generation and cmake patching and get
+        # the possibility to go for to pkg_config based dependency discovery instead.
+        # At the moment, even with the linked work-around, the linkage is mixed-up
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    def _patch_sources(self):
+        # patch 0001 and 0002 are uniquely appliable, based ony dependency config
+        patches_to_apply = ["0003-exclude-unnecessary-cmake-subprojects.patch"]
+        if self.options.with_crypto == "openssl":
+            patches_to_apply.append("0002-libqpdf-cmake-deps-jpeg-zlib-openssl.patch")
+        else:
+            patches_to_apply.append("0001-libqpdf-cmake-deps-jpeg-zlib.patch")
+        for patch_file in patches_to_apply:
+            patch(self, base_path=self.source_folder,
+                  patch_file=os.path.join(self.source_folder, "..", "patches", patch_file))
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE.txt", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        cmake = CMake(self)
+        cmake.install()
+
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
+
+    def package_info(self):
+        self.cpp_info.libs = ["qpdf"]
+        self.cpp_info.set_property("cmake_file_name", "qpdf")
+        self.cpp_info.set_property("cmake_target_name", "qpdf::libqpdf")
+        self.cpp_info.set_property("pkg_config_name", "libqpdf")
+
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.cpp_info.system_libs.append("m")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.filenames["cmake_find_package"] = "qpdf"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "qpdf"
+        self.cpp_info.names["cmake_find_package"] = "qpdf"
+        self.cpp_info.names["cmake_find_package_multi"] = "qpdf"
```

Comment:
```suggestion
    def self.package_info(self):
        self.cpp_info.set_property("cmake_file_name", "qpdf")

        self.cpp_info.components["libqpdf"].libs = ["qpdf"]
        self.cpp_info.components["libqpdf"].set_property("pkg_config_name", "libqpdf")
        self.cpp_info.components["libqpdf"].set_property("cmake_target_name", "qpdf::libqpdf")
        self.cpp_info.components["libqpdf"].requires.append("zlib::zlib")
        self.cpp_info.components["libqpdf"].requires.append(f"{self.options.with_jpeg}::{self.options.with_jpeg}")

        if self.options.with_crypto == "openssl":
            self.cpp_info.components["libqpdf"].requires.append("openssl::openssl")

        if self.settings.os in ["Linux", "FreeBSD"]:
            self.cpp_info.components["libqpdf"].system_libs.append("m")

        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
        self.cpp_info.filenames["cmake_find_package"] = "qpdf"
        self.cpp_info.filenames["cmake_find_package_multi"] = "qpdf"
        self.cpp_info.names["cmake_find_package"] = "qpdf"
        self.cpp_info.names["cmake_find_package_multi"] = "qpdf"
        self.cpp_info.components["libqpdf"].names["cmake_find_package"] = "libqpdf"
        self.cpp_info.components["libqpdf"].names["cmake_find_package_multi"] = "libqpdf"
```

Reply:
And of course I typoed after I copied what I tested locally.... sorry about that.